File: ToDoList.Server/Models/Task.cs
﻿namespace ToDoList.Server.Models
{
    public class Task
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public int CategoryId { get; set; }
        public string? Category { get; set; }
        public DateTime? DueDate { get; set; }
        public bool IsDone { get; set; }
    }
}

File: todolist.client/src/redux/types/task.ts
export type Task = {
  id: number;
  title: string;
  dueDate: string;
  categoryId: number;
  isDone: boolean;
};

export type NewTask = {
  title: string;
  dueDate?: Date;
  categoryId: number;
};

export type DeleteTaskInputType = {
  id: number;
};

export type ToggleTaskInputType = {
  id: number;
};
